exercises/flatten-array/example.sml

```diff
@@ -1,5 +1,5 @@
 datatype 'a nestedList =
-	  Leaf of 'a			     (* leaf *)
+	  Leaf of 'a			           (* leaf *)
```

Comment:
`Leaf` and `Node` make me think about a tree. I think there's a better name maybe something like:
```
datatype 'a nestedList = Elem of 'a | List of 'a nestedList list
```